osu.Game/Screens/OsuScreen.cs

```diff
@@ -120,12 +152,28 @@ public override void OnResuming(IScreen last)
             sampleExit?.Play();
             applyArrivingDefaults(true);
 
+            if (api != null)
+                api.LocalUser.Value.Status.ValueChanged += userStatusChanged;
+
+            ScreenStatus = ScreenStatus;
+
             base.OnResuming(last);
         }
 
+        private void userStatusChanged(ValueChangedEvent<UserStatus> obj)
```

Comment:
this whole thing is ugly. please do as i said and split the whole "current activity" into a separate data structure which doesn't conflict (and require all this extra logic) with online status (online/dnd/invisible).